tests/integration/metrics/test_integration_metric.py

```diff
@@ -0,0 +1,152 @@
+#     Copyright 2020 Dynatrace LLC
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+
+
+import base64
+import os
+from typing import NewType, Any, Dict
+
+import pytest
+from pytest_mock import MockerFixture
+from wiremock.constants import Config
+from wiremock.resources.near_misses import NearMissMatchPatternRequest
+from wiremock.resources.requests import RequestResponseFindResponse, RequestResponseRequest
+from wiremock.resources.requests.resource import Requests
+from wiremock.server import WireMockServer
+
+import lib.credentials
+import lib.entities.extractors.cloud_sql
+import lib.entities.extractors.gce_instance
+import lib.entities.google_api
+import lib.metric_ingest
+from lib.clientsession_provider import init_gcp_client_session
+from lib.context import LoggingContext
+from lib.credentials import fetch_dynatrace_api_key, create_token
+from main import async_dynatrace_gcp_extension
+from assertpy import assert_that
+
+AUTHORIZATION_KEY = b"Fake secret - Open sesame"
+METRIC_MESSAGE_DATA = '{}'
+MOCKED_API_PORT = 9182
+
+MonkeyPatchFixture = NewType("MonkeyPatchFixture", Any)
+system_variables: Dict = {
+    'DYNATRACE_URL': 'http://localhost:' + str(MOCKED_API_PORT),
+    'REQUIRE_VALID_CERTIFICATE': 'False'
+    # 'DYNATRACE_ACCESS_KEY': ACCESS_KEY, this one is encoded in mocks files
+    # 'GCP_SERVICES': "gce_instance/default,api/default,gce_instance/default,cloudsql_database/default"
+}
+
+
+@pytest.fixture(scope="package", autouse=True)
+def setup_test_variables(resource_path_root):
+    lib.credentials._CLOUD_RESOURCE_MANAGER_ROOT = f"http://localhost:{MOCKED_API_PORT}/v1"
+    lib.credentials._SECRET_ROOT = f"http://localhost:{MOCKED_API_PORT}/v1"
+    lib.metric_ingest._MONITORING_ROOT = f"http://localhost:{MOCKED_API_PORT}/v3"
+    lib.entities.extractors.cloud_sql._SQL_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+    lib.entities.google_api._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+    lib.entities.extractors.gce_instance._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+
+    system_variables.update({
```

Comment:
why not simple `system_variables["GOOGLE_APPLICATION_CREDENTIALS"] = f"{resource_path_root}/metrics/token_for_tests.json"`?

Reply:
Ain't that an obvious one :smile:. I'll change that.